browser/ui/webui/brave_wallet/trezor/trezor_ui.cc

```diff
@@ -7,15 +7,70 @@
 
 #include <string>
 
+#include "base/memory/ref_counted_memory.h"
+#include "base/notreached.h"
 #include "brave/components/constants/webui_url_constants.h"
+#include "brave/components/trezor_bridge/resources/grit/trezor_bridge_generated.h"
 #include "brave/components/trezor_bridge/resources/grit/trezor_bridge_generated_map.h"
 #include "components/grit/brave_components_resources.h"
 #include "content/public/browser/web_contents.h"
 #include "content/public/browser/web_ui_data_source.h"
+#include "ui/base/resource/resource_bundle.h"
 #include "ui/webui/resources/grit/webui_resources.h"
 
 namespace {
 constexpr char kTrezorConnectURL[] = "https://connect.trezor.io/";
+
+// Trezor popup page is opened with window.open(url) which doesn't
+// work(opened window loses opener) from chrome-untrusted trezor bridge. We need
+// to revert to legacy behavior which works: open blank page then navigate.
+// So effectively we rollback this change:
+// https://github.com/trezor/trezor-suite/pull/10975/changes#diff-38dd02260cff108b8329d6f3adbf8717b8e8737222de87a92c048f8bbf0bf159R256-R258
+// by overwriting
+// window.open(url, ...args)
+// with
+// window.open('', ...args).location.href = url
+constexpr char kTrezorBundlePatch[] =
+    "(()=>{"
+    " window.open = new Proxy(window.open, {"
+    "  apply(target, thisArg, [url, ...rest]) {"
+    "   const result = Reflect.apply(target, thisArg, ['', ...rest]);"
+    "   if (result && url) {"
+    "    result.location.href = url;"
+    "   }"
+    "   return result;"
+    "  }"
+    " })"
+    "})();\n";
```

Comment:
I might be missing something but why not just place this monkeypatch in trezor.html, somewhere before `IDR_TREZOR_BRIDGE_TREZOR_BUNDLE_JS` is requested, instead of all this C++ work?